BUG: Drift Detection Client ITs fail if service is cached

## Description
If `foundation-drift-detection-service` is skipped due to the Maven build cache, the ITs for `foundation-drift-detection-java-client` fail. This seems to have surfaced due to the build-cache config changes in #653. This would also happen if you had no build cache and tried to build the client module without `-am`.

## Steps to Reproduce
1. Build the service so it is cached
    ```sh
    mvn clean install -Pintegration-test -pl :foundation-drift-detection-service -am
    ```
2. Build the Java client
    ```sh
    mvn clean install -Pintegration-test -pl :foundation-drift-detection-client-java -am
    ```

### Expected Behavior
The ITs pass

### Actual Behavior
The Quarkus app does not start resulting in a _"Connection refused"_ error.

## Definition of Done
- [x] `foundation-drift-detection-client-java` ITs pass even if the service has not been built
  - Use existing pattern for packaging Quarkus apps that we use for containerization

## Test Script
1. Follow the _Steps to Reproduce_
2. Verify the client ITs pass

## Additional Context
Test Results:
```sh
10:05:38 [INFO] Results:
10:05:38 [INFO]
10:05:38 [ERROR] Errors:
10:05:38 [ERROR]   DriftDetectionRestClientIT.testDriftInvocationEndpointWithASingleDriftVariable:59->i_invoke_drift_specifying_the_policy_using_the_rest_service_with_a_single_drift_variable:74->invokeDrift:83 » Processing RESTEASY004655: Unable to invoke request: org.apache.http.conn.HttpHostConnectException: Connect to localhost:8080 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused
10:05:38 [ERROR]   DriftDetectionRestClientIT.testDriftInvocationEndpointWithMultipleDriftVariables:52->i_invoke_drift_specifying_the_policy_using_the_rest_service_with_multiple_drift_variables:69->invokeDrift:83 » Processing RESTEASY004655: Unable to invoke request: org.apache.http.conn.HttpHostConnectException: Connect to localhost:8080 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused
10:05:38 [INFO]
10:05:38 [ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0
```

--- a/foundation/foundation-drift-detection/foundation-drift-detection-client-java/pom.xml
+++ b/foundation/foundation-drift-detection/foundation-drift-detection-client-java/pom.xml
@@ -78,6 +78,8 @@
             <groupId>com.boozallen.aissemble</groupId>
             <artifactId>foundation-drift-detection-service</artifactId>
             <version>${project.version}</version>
+            <classifier>app</classifier>
+            <type>zip</type>
             <scope>test</scope>
         </dependency>
         <dependency>
@@ -123,6 +125,24 @@
             <build>
                 <plugins>
                     <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-dependency-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>unpack-service-app</id>
+                                <phase>pre-integration-test</phase>
+                                <goals>
+                                    <goal>unpack-dependencies</goal>
+                                </goals>
+                                <configuration>
+                                    <outputDirectory>${project.build.directory}</outputDirectory>
+                                    <failOnMissingClassifierArtifact>true</failOnMissingClassifierArtifact>
+                                    <includeClassifiers>app</includeClassifiers>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
                         <groupId>com.bazaarvoice.maven.plugins</groupId>
                         <artifactId>process-exec-maven-plugin</artifactId>
                         <version>0.7</version>
@@ -135,7 +155,6 @@
                                 </goals>
                                 <configuration>
                                     <name>DriftDetectionService</name>
-                                    <workingDir>target</workingDir>
                                     <waitForInterrupt>false</waitForInterrupt>
                                     <healthcheckUrl>http://localhost:8080/invoke-drift/healthcheck</healthcheckUrl>
                                     <arguments>
@@ -143,7 +162,7 @@
                                         <argument>-DKRAUSENING_BASE=${basedir}/src/test/resources/krausening/base</argument>
                                         <argument>-DDRIFT_DETECTION_POLICIES=${basedir}/src/test/resources/drift-policies</argument>
                                         <argument>-jar</argument>
-                                        <argument>${basedir}/../foundation-drift-detection-service/target/quarkus-app/quarkus-run.jar</argument>
+                                        <argument>${project.build.directory}/quarkus-app/quarkus-run.jar</argument>
                                     </arguments>
                                 </configuration>
                             </execution>

--- a/foundation/foundation-drift-detection/foundation-drift-detection-service/pom.xml
+++ b/foundation/foundation-drift-detection/foundation-drift-detection-service/pom.xml
@@ -126,6 +126,26 @@
                 <artifactId>quarkus-maven-plugin</artifactId>
                 <extensions>true</extensions>
             </plugin>
+            <!-- Packages the quarkus app and its dependencies into a zip-->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-assembly-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>single</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <descriptors>
+                        <descriptor>${project.basedir}/src/main/resources/quarkus-app.xml</descriptor>
+                    </descriptors>
+                    <finalName>quarkus</finalName>
+                    <appendAssemblyId>true</appendAssemblyId>
+                </configuration>
+            </plugin>
             <plugin>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>

--- a/foundation/foundation-drift-detection/foundation-drift-detection-service/src/main/resources/quarkus-app.xml
+++ b/foundation/foundation-drift-detection/foundation-drift-detection-service/src/main/resources/quarkus-app.xml
@@ -1,0 +1,29 @@
+<!--
+  #%L
+  AIOps Docker Baseline::Versioning::Service
+  %%
+  Copyright (C) 2021 Booz Allen
+  %%
+  This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+  #L%
+  -->
+<assembly
+    xmlns="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.2"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.2 http://maven.apache.org/xsd/assembly-1.1.2.xsd">
+    <id>app</id>
+
+    <includeBaseDirectory>false</includeBaseDirectory>
+
+    <formats>
+        <format>zip</format>
+    </formats>
+
+    <fileSets>
+        <fileSet>
+            <directory>${project.build.directory}/quarkus-app/</directory>
+            <outputDirectory>quarkus-app/</outputDirectory>
+        </fileSet>
+    </fileSets>
+
+</assembly>

